include/util/xor_fast_hash.hpp

```diff
@@ -29,64 +37,30 @@ namespace util
 */
 class XORFastHash
 { // 65k entries
-    std::vector<unsigned short> table1;
-    std::vector<unsigned short> table2;
+    std::array<std::uint16_t, (1u << 16u)> table1;
```

Comment:
Hmm the max size also depends on the `uint32_t` type in the `operator()` and how we split it into two `uint16_t` there. That is, the highest index a `uint16_t` can hold is `65535`, therefore we are only able to index `65536` items in each table.

 Therefore if we template the max size but keep the `uint32_t` for hashable types, we should at least

    static_assert(MaxSize <= (1u << 16u), "");